File: Bedu/Java/Clase4/AnimalesSonidos/app/src/main/java/com/charlye934/animalessonidos/MainActivity.java
package com.charlye934.animalessonidos;

import androidx.appcompat.app.AppCompatActivity;

import android.media.MediaPlayer;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

import java.util.HashMap;
import java.util.Map;

public class MainActivity extends AppCompatActivity {

    Map<String, Integer> animalesList = new HashMap<>();
    MediaPlayer sonido;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        
        animalesList.put("caballo", R.raw.caballo);
        animalesList.put("cabra", R.raw.cabra);
        animalesList.put("cerdo", R.raw.cerdo);
        animalesList.put("gallo", R.raw.gallo);
        animalesList.put("mono", R.raw.mono);
        animalesList.put("perro", R.raw.perro);
        animalesList.put("serpiente", R.raw.serpiente);
        animalesList.put("tigre", R.raw.tigre);
    }

    public void ejecutaSonido(String nombreAnimal){
        int ani = animalesList.get(nombreAnimal);
        if(sonido != null){
            sonido.stop();
        }
        sonido = MediaPlayer.create(getApplicationContext(), ani);
        sonido.start();

    }

    public void seleccionaAnimal(View view){
        ImageView v = (ImageView)view;
        String animal = String.valueOf(v.getTag());
        ejecutaSonido(animal);
    }
}

File: Bedu/Clase1-4/AnimalesSonidosKotlin/app/src/main/java/com/charlye934/animalessonidoskotlin/MainActivity.kt
package com.charlye934.animalessonidoskotlin

import android.media.MediaPlayer
import android.os.Bundle
import android.view.View
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import java.util.*

class MainActivity : AppCompatActivity() {

    var animalesList: MutableMap<String, Int> = HashMap()
    var sonido: MediaPlayer? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        animalesList["caballo"] = R.raw.caballo
        animalesList["cabra"] = R.raw.cabra
        animalesList["cerdo"] = R.raw.cerdo
        animalesList["gallo"] = R.raw.gallo
        animalesList["mono"] = R.raw.mono
        animalesList["perro"] = R.raw.perro
        animalesList["serpiente"] = R.raw.serpiente
        animalesList["tigre"] = R.raw.tigre
    }

    fun ejecutaSonido(nombreAnimal: String) {
        val ani: Int? = animalesList.get(nombreAnimal)
        if (sonido != null) {
            sonido!!.stop()
        }
        if(ani != null) MediaPlayer.create(applicationContext, ani).start()
    }

    fun seleccionaAnimal(view: View) {
        val v = view as ImageView
        val animal = v.tag.toString()
        ejecutaSonido(animal)
    }
}
